api/users/serializers.py

```diff
@@ -142,6 +172,13 @@ def update(self, instance, validated_data):
                                 detail='{} only accept a list of one single value'. format(key)
                             )
                         instance.social[key] = val[0]
+            elif 'schools' == attr:
+                self._validate_user_json(value, 'education-schema.json')
```

Comment:
In order to reduce the amount of logic that happens in this method (and better separate logic), you could implement [field level validators](http://www.django-rest-framework.org/api-guide/serializers/#field-level-validation) for `schools` and `jobs` (each method would just call `self._validate_user_json(...)`). 

See @erinspace's work in #8551 for an example of this.